goth_tests/domain/ya-provider/test_provider_multi_activity.py

```diff
@@ -91,3 +101,67 @@ async def test_provider_multi_activity(
 
         # Payment
         await pay_all(requestor, agreement_providers)
+
+
+@pytest.mark.asyncio
+async def test_provider_single_activity_at_once(
+    common_assets: Path,
+    config_overrides: List[Override],
+    log_dir: Path,
+):
+    """Test provider rejecting second activity if one is already running.
+
+    Provider is expected to reject second activity, if one is already running.
+    """
+    runner, config = _create_runner(common_assets, config_overrides, log_dir)
+
+    async with runner(config.containers):
+        requestor = runner.get_probes(probe_type=RequestorProbe)[0]
+        providers = runner.get_probes(probe_type=ProviderProbe)
+
+        # Market
+        task_package = wasi_task_package.format(
+            web_server_addr=runner.host_address, web_server_port=runner.web_server_port
+        )
+
+        demand = (
+            DemandBuilder(requestor)
+            .props_from_template(task_package)
+            .property("golem.srv.caps.multi-activity", True)
+            .constraints(
+                "(&(golem.com.pricing.model=linear)\
+                (golem.srv.caps.multi-activity=true)\
+                (golem.runtime.name=wasmtime))"
+            )
+            .build()
+        )
+
+        agreement_providers = await negotiate_agreements(
+            requestor,
+            demand,
+            providers,
+        )
+
+        #  Activity
+        agreement_id, provider = agreement_providers[0]
+
+        activity_id1 = await requestor.create_activity(agreement_id)
+
+        # Creation should fail here.
+        with pytest.raises(ApiException) as e:
+            await requestor.create_activity(agreement_id)
+
+            assert (
+                re.match(
+                    r"terminated. Reason: Only single Activity allowed,"
+                    r" message: Can't create 2 simultaneous Activities.",
+                    e.value.body,
+                )
+                is not None
+            )
```

Comment:
```suggestion
        with pytest.raises(ApiException) as e:
            await requestor.create_activity(agreement_id)

        assert (
            re.match(
                r"terminated. Reason: Only single Activity allowed,"
                r" message: Can't create 2 simultaneous Activities.",
                e.value.body,
            )
        )
```
If there's an error in `requestor.create_activity(agreement_id)` then the control exits the scope of `with pytest.raises(...)` and the assertion is not checked.

Reply:
Thanks, good catch :)

Reply:
Good you noticed, because this regex was invalid and failed to match ;)